jlm/hls/backend/rvsdg2rhls/UnusedStateRemoval.cpp

```diff
@@ -109,12 +109,9 @@ RemoveUnusedStatesFromLambda(llvm::lambda::node & lambdaNode)
 
   JLM_ASSERT(lambdaNode.output()->nusers() == 1);
   lambdaNode.region()->RemoveResult((*lambdaNode.output()->begin())->index());
+  auto oldExport = lambdaNode.ComputeCallSummary()->GetRvsdgExport();
+  jlm::llvm::GraphExport::Create(*newLambdaOutput, oldExport ? oldExport->Name() : "");
```

Comment:
The GraphExport moves functionality from rvsdg to llvm, making the hls dependence on llvm more explicit. But I guess there's no benefit of trying to keep it more decoupled, as it anyway depends on the lambda.

This made me think about the naming conventions that are used.

The hls backend, of course, takes rvsdg-llvm as input and has a direct dependence on it. This is not reflected in the rvsdg2rhls naming. It maybe would be more logical to call it llvm2rhls, but that has the disadvantage that it can sound like it's llvm-IR and not rvsdg-llvm.
Potentially more informative naming could be: rvsdg-llvm2rvsdg-hls, r-llvm2r-hls, rllvm2rhls.

Similar naming ambiguity exists in the llvm backend with rvsdg2jlm and jlm2llvm, which could sound like the generic rvsdg 'dialect' (jlm/rvsdg) is taken as input. It also makes me wonder why jlm is the name of the intermediate step, as this is the name of the whole compiler. Wouldn't a different, more descriptive, name be suitable here?

Reply:
Hls is dependent on the LLVM dialect anyway. It would not work without it in the first place. Thus, this change does not introduce any unnecessary dependencies (that are not already there and required).

All these "sub-"namespaces should disappear and they class/methods should express their name properly. However, this is for another PR.